### PR TITLE
app-misc/mosquitto: bump to 1.6.2

### PR DESCRIPTION
**Pull request by oxr463**
Package-Manager: Portage-2.3.66, Repoman-2.3.12
Signed-off-by: Lucas Ramage <ramage.lucas@protonmail.com>

### CODE REVIEW

**Comment by oxr463** [comment]
> ```
> distcc[54] ERROR: compile (null) on localhost failed
> persist_read_test.c:6:10: fatal error: CUnit/CUnit.h: No such file or directory
>  #include <CUnit/CUnit.h>
>           ^~~~~~~~~~~~~~~
> compilation terminated.
> distcc[52] ERROR: compile persist_read_test.c on localhost failed
> make[2]: *** [<builtin>: persist_read_test.o] Error 1
> ```

Ping @mgorny, please check again.

**Comment by mgorny** [comment]
```
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/mosquitto_pub
 * /usr/bin/mosquitto_rr
 * /usr/bin/mosquitto_sub

**Comment by mgorny** [comment]
```
RepoMan scours the neighborhood...
  RDEPEND.suspect               1
   app-misc/mosquitto/mosquitto-1.6.2.ebuild: 'test?' USE conditional in RDEPEND

**Comment by oxr463** [comment]
> ```
>  * QA Notice: Files built without respecting LDFLAGS have been detected
>  *  Please include the following list of files in your report:
>  * /usr/bin/mosquitto_pub
>  * /usr/bin/mosquitto_rr
>  * /usr/bin/mosquitto_sub
> ```

How did you get this output?

I'm running this command to install:

```terminal
FEATURES="test" ACCEPT_KEYWORDS="~amd64" USE="bridge" emerge app-misc/mosquitto
```

> ```
> RepoMan scours the neighborhood...
>   RDEPEND.suspect               1
>    app-misc/mosquitto/mosquitto-1.6.2.ebuild: 'test?' USE conditional in RDEPEND
> ```

Oops, I'll move that into `DEPEND` directly.

**Comment by mgorny** [comment]
On Thu, 2019-05-23 at 15:20 -0700, rage wrote:
> > ```
> >  * QA Notice: Files built without respecting LDFLAGS have been detected
> >  *  Please include the following list of files in your report:
> >  * /usr/bin/mosquitto_pub
> >  * /usr/bin/mosquitto_rr
> >  * /usr/bin/mosquitto_sub
> > ```
> 
> How did you get this output?

`LDFLAGS="-Wl,--defsym=__gentoo_check_ldflags__=0"`

**Comment by oxr463** [comment]
https://github.com/eclipse/mosquitto/commit/3b4a5419c34d3b1f732bf4bf5aed44fd21c8c1d0

**Comment by oxr463** [comment]
Created an issue upstream, (See: https://github.com/eclipse/mosquitto/issues/1294).

**Comment by oxr463** [comment]
@mgorny please review again. Thank you

**Comment by oxr463** [comment]
Ping @mgorny

File: app-misc/mosquitto/mosquitto-1.6.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_6 )
```
Comment (mgorny) [inline app-misc/mosquitto/mosquitto-1.6.2.ebuild]: 3.7?

File: app-misc/mosquitto/mosquitto-1.6.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+SRC_URI="https://mosquitto.org/files/source/${P}.tar.gz"
+
+LICENSE="EPL-1.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE="bridge examples +persistence +srv ssl tcpd test websockets"
+
+REQUIRED_USE="test? ( bridge )"
+
+RDEPEND="srv? ( net-dns/c-ares )
```
Comment (mgorny) [inline app-misc/mosquitto/mosquitto-1.6.2.ebuild]: `:=`, it's using subslots.

File: app-misc/mosquitto/mosquitto-1.6.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	ssl? ( dev-libs/openssl:0= )
+	tcpd? ( sys-apps/tcp-wrappers )"
+
+DEPEND="${PYTHON_DEPS}
+	${RDEPEND}
+	test? ( dev-util/cunit )
+	websockets? ( net-libs/libwebsockets )"
+
+_emake() {
+	LIBDIR=$(get_libdir)
```
Comment (mgorny) [inline app-misc/mosquitto/mosquitto-1.6.2.ebuild]: Why do you need a global variable here?

**Comment by oxr463** [comment]
@mgorny, I made the requested changes, however, the test-suite is failing for me due to a kernel upgrade, (I think?), https://github.com/eclipse/mosquitto/issues/1307.

**Comment by mgorny** [comment]
I get:
```
./11-message-expiry.py
1559979290: mosquitto version 1.6.2 starting
1559979290: Config loaded from 11-message-expiry.conf.
1559979290: Opening ipv4 listen socket on port 1888.
1559979290: Opening ipv6 listen socket on port 1888.
1559979290: New connection from ::1 on port 1888.
1559979290: Socket error on client <unknown>, disconnecting.
1559979292: New connection from 127.0.0.1 on port 1888.
1559979292: New client connected from 127.0.0.1 as subpub-qos0-test (p5, c0, k60).
1559979292: No will message specified.
1559979292: Sending CONNACK to subpub-qos0-test (1, 0)
1559979292: Sending PUBLISH to subpub-qos0-test (d0, q1, r0, m1, 'subpub/qos1', ... (8 bytes))
1559979292: Sending PUBLISH to subpub-qos0-test (d0, q1, r0, m2, 'subpub/qos1', ... (8 bytes))
1559979292: Sending PUBLISH to subpub-qos0-test (d0, q1, r0, m3, 'subpub/qos1', ... (8 bytes))
1559979292: Socket error on client subpub-qos0-test, disconnecting.
1559979292: mosquitto version 1.6.2 terminating
1559979292: Saving in-memory database to mosquitto-1888.db.
```

Maybe run it a few times to see if it's flakiness.

**Comment by oxr463** [comment]
> I get:
> 
> ```
> ./11-message-expiry.py
> 1559979290: mosquitto version 1.6.2 starting
> 1559979290: Config loaded from 11-message-expiry.conf.
> 1559979290: Opening ipv4 listen socket on port 1888.
> 1559979290: Opening ipv6 listen socket on port 1888.
> 1559979290: New connection from ::1 on port 1888.
> 1559979290: Socket error on client <unknown>, disconnecting.
> 1559979292: New connection from 127.0.0.1 on port 1888.
> 1559979292: New client connected from 127.0.0.1 as subpub-qos0-test (p5, c0, k60).
> 1559979292: No will message specified.
> 1559979292: Sending CONNACK to subpub-qos0-test (1, 0)
> 1559979292: Sending PUBLISH to subpub-qos0-test (d0, q1, r0, m1, 'subpub/qos1', ... (8 bytes))
> 1559979292: Sending PUBLISH to subpub-qos0-test (d0, q1, r0, m2, 'subpub/qos1', ... (8 bytes))
> 1559979292: Sending PUBLISH to subpub-qos0-test (d0, q1, r0, m3, 'subpub/qos1', ... (8 bytes))
> 1559979292: Socket error on client subpub-qos0-test, disconnecting.
> 1559979292: mosquitto version 1.6.2 terminating
> 1559979292: Saving in-memory database to mosquitto-1888.db.
> ```
> 
> Maybe run it a few times to see if it's flakiness.

I tried it again in a virtual machine running Linux kernel 4.19, and I'm getting the same error as before so I don't think it has anything to do with my kernel version.

**Comment by oxr463** [comment]
@mgorny, please review. I removed all failing tests from the Makefile.

**Comment by mgorny** [comment]
Do you have any clue why they fail? Is upstream aware?

**Comment by oxr463** [comment]
> Do you have any clue why they fail? Is upstream aware?

I'm not 100% sure why they are failing, but I created an issue upstream; I'll continue troubleshooting from there.

**Pull request closed without merging**